File: App_Plugins/HelloWorld/Scripts/HelloWorldController.js
angular.module('umbraco').controller("HelloWorldController", ["$scope", "$http", function($scope, $http) {

    $scope.AjaxData = '';

    // backoffice/{PluginController Area}/{Controller}/{Action}
    $http.get('backoffice/helloWorldApi/HelloWorld/GetContent').then(
        function(response) {
            console.log(response);

            $scope.AjaxData = response.data.Data;
        }, 
        function() { console.log('Error'); });
}]);

//angular.module('umbraco').controller("HelloWorldController", HelloWorldModule.HelloWorldController);

File: App_Plugins/HelloWorld/Controllers/HelloWorldController.cs
namespace umbraco_demo.App_Plugins.HelloWorld.Controllers
{
    using Umbraco.Web.Editors;
    using Umbraco.Web.Mvc;

    [PluginController("helloWorldApi")]
    public class HelloWorldController : UmbracoAuthorizedJsonController
    {
        // GET: GetContent
        // If the get method doesn't start with the word get you get a 405 for some reason.
        public ActionResult GetContent()
        {
            return new JsonResult
            {
                Data = "Hello Umbraco!",
                ContentType = "application/json"
            };
        }
    }
}
